PWGHF/D2H/TableProducer/candidateCreatorBplusReduced.cxx

```diff
@@ -93,11 +100,114 @@ struct HfCandidateCreatorBplusReduced {
+    auto covMatrixPV = primaryVertex.getCov();
+
+    // Set the magnetic field from ccdb
+    bz = collision.bz();
+    df2.setBz(bz);
+
+    for (const auto& candD0 : candsDThisColl) {
+      auto trackParCovD = getTrackParCov(candD0);
+      std::array<float, 3> pVecD0 = candD0.pVector();
+
+      for (const auto& trackPion : tracksPionThisCollision) {
+        auto trackParCovPi = getTrackParCov(trackPion);
+        std::array<float, 3> pVecPion = trackPion.pVector();
+
+        // compute invariant mass square and apply selection
+        auto invMass2D0Pi = RecoDecay::m2(std::array{pVecD0, pVecPion}, std::array{massD0, massPi});
+        if ((invMass2D0Pi < invMass2D0PiMin) || (invMass2D0Pi > invMass2D0PiMax)) {
+          continue;
+        }
+        // ---------------------------------
+        // reconstruct the 2-prong B+ vertex
+        hCandidates->Fill(SVFitting::BeforeFit);
+        try {
+          if (df2.process(trackParCovD, trackParCovPi) == 0) {
+            continue;
+          }
+        } catch (const std::runtime_error& error) {
+          LOG(info) << "Run time error found: " << error.what() << ". DCFitterN cannot work, skipping the candidate.";
+          hCandidates->Fill(SVFitting::Fail);
+          continue;
+        }
+        hCandidates->Fill(SVFitting::FitOk);
+        // D0Pi passed B+ reconstruction
+
+        // calculate relevant properties
+        const auto& secondaryVertexBplus = df2.getPCACandidate();
+        auto chi2PCA = df2.getChi2AtPCACandidate();
+        auto covMatrixPCA = df2.calcPCACovMatrixFlat();
+        registry.fill(HIST("hCovSVXX"), covMatrixPCA[0]);
+        registry.fill(HIST("hCovPVXX"), covMatrixPV[0]);
+
+        // propagate D0 and Pi to the B+ vertex
+        df2.propagateTracksToVertex();
+        // track.getPxPyPzGlo(pVec) modifies pVec of track
+        df2.getTrack(0).getPxPyPzGlo(pVecD0);   // momentum of D0 at the B+ vertex
+        df2.getTrack(1).getPxPyPzGlo(pVecPion); // momentum of Pi at the B+ vertex
+
+        registry.fill(HIST("hMassBplusToD0Pi"), std::sqrt(invMass2D0Pi));
+
+        // compute impact parameters of D0 and Pi
+        o2::dataformats::DCA dcaD0;
+        o2::dataformats::DCA dcaPion;
+        trackParCovD.propagateToDCA(primaryVertex, bz, &dcaD0);
+        trackParCovPi.propagateToDCA(primaryVertex, bz, &dcaPion);
+
+        // get uncertainty of the decay length
+        double phi, theta;
+        // getPointDirection modifies phi and theta
+        getPointDirection(std::array{collision.posX(), collision.posY(), collision.posZ()}, secondaryVertexBplus, phi, theta);
+        auto errorDecayLength = std::sqrt(getRotatedCovMatrixXX(covMatrixPV, phi, theta) + getRotatedCovMatrixXX(covMatrixPCA, phi, theta));
+        auto errorDecayLengthXY = std::sqrt(getRotatedCovMatrixXX(covMatrixPV, phi, 0.) + getRotatedCovMatrixXX(covMatrixPCA, phi, 0.));
+
+        int hfFlag = BIT(hf_cand_bplus::DecayType::BplusToD0Pi);
```

Comment:
I would say that we do not need to store `hfFlag` since only one channel is possible (we should also remove it for the B0 actually)